lib/Sema/CSGen.cpp

```diff
@@ -4409,6 +4409,13 @@ swift::resolveValueMember(DeclContext &DC, Type BaseTy, DeclName Name) {
   if (LookupResult.ViableCandidates.empty())
     return Result;
 
+  // If there's only one viable member, that is the best one.
+  if (LookupResult.ViableCandidates.size() == 1) {
+    Result.Impl->BestIdx = Result.Impl->AllDecls.size();
+    Result.Impl->AllDecls.push_back(LookupResult.ViableCandidates[0].getDecl());
+    return Result;
+  }
+
   // Try to figure out the best overload.
   ConstraintLocator *Locator = CS.getConstraintLocator({});
```

Comment:
Constraint locators can now be created by decls. This should be updated it to point something relevant. We will fix it in separate PR targeting `master` only.

Reply:
That would be great, thank you!